internal/cephfs/cephfs_util.go

```diff
@@ -46,38 +46,26 @@ func (vo *volumeOptions) getFscID(ctx context.Context) (int64, error) {
 	return 0, ErrVolumeNotFound
 }
 
-// CephFilesystem is a representation of the json structure returned by 'ceph fs ls'.
-type CephFilesystem struct {
-	Name           string   `json:"name"`
-	MetadataPool   string   `json:"metadata_pool"`
-	MetadataPoolID int      `json:"metadata_pool_id"`
-	DataPools      []string `json:"data_pools"`
-	DataPoolIDs    []int    `json:"data_pool_ids"`
-}
+func (vo *volumeOptions) getMetadataPool(ctx context.Context) (string, error) {
+	fsa, err := vo.conn.GetFSAdmin()
+	if err != nil {
+		util.ErrorLog(ctx, "could not get FSAdmin, can not fetch metadata pool for %s:", vo.FsName, err)
+		return "", err
+	}
 
-func getMetadataPool(ctx context.Context, monitors string, cr *util.Credentials, fsName string) (string, error) {
-	// ./tbox ceph fs ls --format=json
-	// [{"name":"myfs","metadata_pool":"myfs-metadata","metadata_pool_id":4,...},...]
-	var filesystems []CephFilesystem
-	err := execCommandJSON(ctx, &filesystems,
-		"ceph",
-		"-m", monitors,
-		"--id", cr.ID,
-		"--keyfile="+cr.KeyFile,
-		"-c", util.CephConfigPath,
-		"fs", "ls", "--format=json",
-	)
+	fsPoolInfos, err := fsa.ListFileSystems()
 	if err != nil {
+		util.ErrorLog(ctx, "could not list filesystems, can not fetch metadata pool for %s:", vo.FsName, err)
 		return "", err
```

Comment:
@nixpanic dont we have to defer `fsa` here?

Reply:
no, there is no destructor for FSAdmin

Reply:
oh..ok .. its good to have.. We dont have to leave the connection behind.